libr/debug/p/native/windows/windows_debug.c

```diff
@@ -0,0 +1,1164 @@
+	if (!rio) {
+		eprintf ("w32_init: failed to allocate memory\n");
+		return false;
+	}
+	setup_debug_privileges (true);
+	// rio->dbgpriv = setup_debug_privileges (true);
+	rio->ph = (HANDLE)NULL;
+	rio->debug = false;
+
+	HMODULE lib = GetModuleHandle ("kernel32"); //Always loaded
+	w32_GetEnabledXStateFeatures = GetProcAddress (lib, "GetEnabledXStateFeatures");
+	w32_InitializeContext = GetProcAddress (lib, "InitializeContext");
+	w32_GetXStateFeaturesMask = GetProcAddress (lib, "GetXStateFeaturesMask");
+	w32_LocateXStateFeature = GetProcAddress (lib, "LocateXStateFeature");
+	w32_SetXStateFeaturesMask = GetProcAddress (lib, "SetXStateFeaturesMask");
+
+	lib = GetModuleHandle ("ntdll"); //Always loaded
+	w32_NtQueryInformationThread = GetProcAddress (lib, "NtQueryInformationThread");
+	return true;
+}
+
+static int suspend_thread(HANDLE th, int bits) {
+	int ret;
+	//if (bits == R_SYS_BITS_32) {
+		if ((ret = SuspendThread (th)) == -1) {
+			r_sys_perror ("suspend_thread/SuspendThread");
+		}
+	/*} else {
+		if ((ret = Wow64SuspendThread (th)) == -1) {
+			r_sys_perror ("suspend_thread/Wow64SuspendThread");
+		}
+	}*/
+	return ret;
+}
+
+static int resume_thread(HANDLE th, int bits) {
+	int ret;
+	//if (bits == R_SYS_BITS_32) {
+		if ((ret = ResumeThread (th)) == -1) {
+			r_sys_perror ("resume_thread/ResumeThread");
+		}
+	/*} else {
+		if ((ret = ResumeThread (th)) == -1) {
+			r_sys_perror ("resume_thread/Wow64ResumeThread");
+		}
+	}*/
+	return ret;
+}
+
+static int set_thread_context(HANDLE th, const ut8 *buf, int size, int bits) {
+	bool ret;
+	//if (bits == R_SYS_BITS_32) {
+		CONTEXT ctx = {0};
+		if (size > sizeof (ctx)) {
+			size = sizeof (ctx);
+		}
+		memcpy (&ctx, buf, size);
+		if(!(ret = SetThreadContext (th, &ctx))) {
+			r_sys_perror ("set_thread_context/SetThreadContext");
+		}
+	/*} else {
+		WOW64_CONTEXT ctx = {0};
+		if (size > sizeof (ctx)) {
+			size = sizeof (ctx);
+		}
+		memcpy (&ctx, buf, size);
+		if(!(ret = Wow64SetThreadContext (th, &ctx))) {
+			r_sys_perror ("set_thread_context/Wow64SetThreadContext");
+		}
+	}*/
+	return ret;
+}
+
+static int get_thread_context(HANDLE th, ut8 *buf, int size, int bits) {
+	int ret = 0;
+	//if (bits == R_SYS_BITS_32) {
+		CONTEXT ctx = {0};
+		// TODO: support various types?
+		ctx.ContextFlags = CONTEXT_ALL;
+		if (GetThreadContext (th, &ctx)) {
+			if (size > sizeof (ctx)) {
+				size = sizeof (ctx);
+			}
+			memcpy (buf, &ctx, size);
+			ret = size;
+		} else {
+			r_sys_perror ("get_thread_context/GetThreadContext");
+		}
+	/*} else {
+		WOW64_CONTEXT ctx = {0};
+		// TODO: support various types?
+		ctx.ContextFlags = CONTEXT_ALL;
+		if (Wow64GetThreadContext (th, &ctx)) {
+			if (size > sizeof (ctx)) {
+				size = sizeof (ctx);
+			}
+			memcpy (buf, &ctx, size);
+			ret = size;
+		} else {
+			r_sys_perror ("get_thread_context/Wow64GetThreadContext");
+		}
+	}*/
+	return ret;
+}
+
+static int GetAVX(HANDLE hThread, ut128 xmm[16], ut128 ymm[16]) {
+	BOOL Success;
```

Comment:
uppercases variables are not following the dev rules

Reply:
also for success. i thnk we can just avoid it and use `if (!...)` inistead of asiining a variable many times